WalletWasabi.Tests/UnitTests/JsonConverters/NetworkJsonConverterTests.cs

```diff
@@ -0,0 +1,167 @@
+using NBitcoin;
+using System.Text.Json;
+using WalletWasabi.JsonConverters;
+using Xunit;
+using JsonConvertNew = System.Text.Json.JsonSerializer;
+using JsonConvertOld = Newtonsoft.Json.JsonConvert;
+
+namespace WalletWasabi.Tests.UnitTests.JsonConverters;
+
+/// <summary>
+/// Tests for <see cref="NetworkJsonConverter"/> and <see cref="NetworkJsonConverterNg"/> classes.
+/// </summary>
+public class NetworkJsonConverterTests
+{
+	/// <summary>
+	/// Tests that JSON converter based on <c>System.Text.Json</c> and the one based on <c>NewtonSoft.Json</c> *serialize* objects equally.
+	/// </summary>
+	[Fact]
+	public void SerializationParity()
+	{
+		TestData testObject = new();
+
+		string json = AssertSerializedEqually(testObject);
+		Assert.Equal("""{"Main":"Main","Test":"TestNet","RegTest":"RegTest","Default":null,"NotAnnotated":null}""", json);
+	}
+
+	/// <summary>
+	/// Tests that JSON converter based on <c>System.Text.Json</c> and the one based on <c>NewtonSoft.Json</c> *deserialize* objects equally.
+	/// </summary>
+	[Fact]
+	public void DeserializationParity()
+	{
+		// Success cases.
+		{
+			string token = "Main";
+			AssertBothDeserialize(S(token));
+
+			token = "TestNet";
+			AssertBothDeserialize(S(token));
+
+			token = "RegTest";
+			AssertBothDeserialize(S(token));
+
+			token = "regression";
+			AssertBothDeserialize(S(token)); // Regtest
+
+			token = "Test";
+			AssertBothDeserialize(S(token)); // Testnet
+
+			token = "Reg";
+			AssertBothDeserialize(S(token)); // Regtest
+
+			token = "Man";
+			AssertBothDeserialize(S(token)); // null
+
+			token = " ";
+			AssertBothDeserialize(S(token)); // null
+
+			token = "null";
+			AssertBothDeserialize(S(token)); // null
+		}
+
+		// Failing cases.
+		{
+			string token = "100";
+			AssertDeserializeDifferentExceptions<InvalidCastException, JsonException>(token);
+		}
+
+		// Unique cases.
+		{
+			string token = "null";
```

Comment:
The token `"null"` is part of the "Success cases" and also part of the "Unique cases".
In one case it can be deserialized and it leaves the Network property empty,
in the other case exceptions are thrown.

Which one is the correct behavior?

Reply:
At L72 we send in `null`. At L60 we send in `"null"`. It's different, one is actually `null`, the other is a string value containing null.